PIMbench/cpp-histogram/README.md

```diff
@@ -1,43 +1,109 @@
 
 ## Input Files
 
-As mentioned above, only 24-bit .bmp files can be used to test the functionality of this benchmark. Sample inputs can be found in the `~/cpp-histogram/histogram_datafiles/` directory, which were gathered from the Phoenix [GitHub](https://github.com/fasiddique/DRAMAP-Phoenix/tree/main) ([direct download link](http://csl.stanford.edu/~christos/data/histogram.tar.gz)), with the execepton of `sample1.bmp`, which came from [FileSamplesHub](https://filesampleshub.com/format/image/bmp). Additional files that exceeded the file size for GitHub which were used in benchmarks for the paper can be found in the following Google Drive [folder](https://drive.google.com/drive/u/3/folders/1sKFcEftxzln6rtjftChb5Yog_9S5CDRd).
+As mentioned above, only 24-bit .bmp files can currently be used to test the functionality of this benchmark. Sample inputs can be found in the `/histogram_datafiles/` directory, which were gathered from the Phoenix [GitHub](https://github.com/fasiddique/DRAMAP-Phoenix/tree/main) ([direct download link](http://csl.stanford.edu/~christos/data/histogram.tar.gz)), with the execepton of `sample1.bmp`, which came from [FileSamplesHub](https://filesampleshub.com/format/image/bmp). Additional files that exceeded the file size for GitHub which were used in benchmarks for the paper can be found in the following Google Drive [folder](https://drive.google.com/drive/u/3/folders/1sKFcEftxzln6rtjftChb5Yog_9S5CDRd).
 
-## Compilation Instructions
+## Directory Structure
 
-To compile, simply run:
+```
+cpp-histogram/
+├── PIM/
+│   ├── Makefile
+│   ├── hist.cpp
+├── baselines/
+│   ├── CPU/
+│   │   ├── Makefile
+│   │   ├── hist.cpp
+│   ├── GPU/
+│   │   ├── Makefile
+│   │   ├── hist.cu
+├── histogram_datafiles/
+│   ├── sample1.bmp
+│   ├── small.bmp
+├── README.md
+├── Makefile
+```
+
+## Implementation Description
+
+This repository contains three different implementations of the histogram benchmark:
+
+1. CPU
+2. GPU
+3. PIM
+
+### Baseline Implementation
+
+CPU and GPU have been used as baselines.
+
+#### CPU
+
+The CPU variant of histogram has been implemented using standard C++ library and threading for parallel execution. Other C++ libraries, such as OpenBLAS or Boost, were not used as either histogram implementations did not exist or did fall not fall into the scope of our use case. Threading was chosen in favor of OpenMP as timings had no significant differences, and due to previous works, which we modeled our benchmarks after, using the same method.
```

Comment:
```suggestion
The CPU variant of histogram has been implemented using standard C++ library and pthread for parallel execution. Other C++ libraries, such as OpenBLAS or Boost, were not used as either histogram implementations did not exist or did fall not fall into the scope of our use case. Threading was chosen in favor of OpenMP as timings had no significant differences, and due to previous works, which we modeled our benchmarks after, using the same method.
```
Also mention that this code was taken from phoenix benchmark and was modified accordingly.